libkbfs/root_metadata.go

```diff
@@ -979,56 +979,24 @@ func (irmd ImmutableRootMetadata) LastModifyingWriterVerifyingKey() kbfscrypto.V
 	return irmd.lastWriterVerifyingKey
 }
 
-// RootMetadataSigned is the top-level MD object stored in MD server
-//
-// TODO: Have separate types for:
-//
-// - The in-memory client representation (needs untrustedServerTimestamp);
-// - the type sent over RPC;
-// - the type stored in the journal;
-// - and the type stored in the MD server.
+// RootMetadataSigned is a wrapper around kbfsmd.RootMetadataSigned
+// that adds an untrusted server timestamp.
 type RootMetadataSigned struct {
-	// SigInfo is the signature over the root metadata by the
-	// last modifying user's private signing key.
-	SigInfo kbfscrypto.SignatureInfo
-	// WriterSigInfo is the signature over the writer metadata by
-	// the last modifying writer's private signing key.
-	WriterSigInfo kbfscrypto.SignatureInfo
-	// all the metadata
-	MD BareRootMetadata
+	kbfsmd.RootMetadataSigned
```

Comment:
Why is this embedded and not aliased? The unexported field? You can export it in `kbfsmd.RootMetadataSigned` and exclude it from the codec.

Reply:
Ah, server code doesn't use it. In fact, it deserializes the timestamp separately (since the protocol also splits it out).

Reply:
I'm confused...wouldn't it still be worth it to keep the types the same?

Reply:
Hmm...my thinking is that it's more confusing to have a field that's used only on the client-side, and duplicated on the protocol/server side. Also, the situation is analogous to BareRMD vs. RMD, where RMD used to have only a single unexported field, and then now it has a bunch.

Reply:
Hmm. All right, let's keep it how you've done it.